core/elite_edge_node.go

```diff
@@ -153,19 +179,37 @@ func (a *AggregatedEENVotes) Pick(b *AggregatedEENVotes) (*AggregatedEENVotes, e
 	return a, nil
 }
 
-// Validate verifies the voteset.
-func (a *AggregatedEENVotes) Validate(eenpWithStake *EliteEdgeNodePool) result.Result {
-	if len(a.Multiplies) != eenpWithStake.Len() {
-		return result.Error("multiplies size %d is not equal to eenp size %d", len(a.Multiplies), eenpWithStake.Len())
+// Validate performs basic validation of the voteset.
+func (a *AggregatedEENVotes) Validate(eenp EliteEdgeNodePool) result.Result {
+	if eenp == nil {
+		return result.Error("empty eenp")
 	}
 	if a.Signature == nil {
 		return result.Error("signature cannot be nil")
 	}
-	pubKeys := eenpWithStake.PubKeys()
-	aggPubkey := bls.AggregatePublicKeysVec(pubKeys, a.Multiplies)
+	if len(a.Addresses) == 0 {
+		return result.Error("aggregated vote is empty")
+	}
+	if len(a.Addresses) != a.Abs() || len(a.Addresses) != len(a.Multiplies) {
```

Comment:
EliteEdgeNodeEngine.convertVote() and AggregatedEENVotes.Merge() should always guarantee that len(a.Addresses) == len(a.Multiplies)? Do we need this condition len(a.Addresses) != a.Abs()?

Or in the code there are other places to generate an aggregatedEENVote?